### PR TITLE
Problem: pulp not provisioned during gem publish step

### PR DESCRIPTION
**Pull request by dkliban**
Solution: configure travis environment for gem publish step

re: #4695
https://pulp.plan.io/issues/4695

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -35,6 +45,9 @@ jobs:
         secure: pQVElIk7hM0cyyGz8K+0R23OlcSBocYibi7yOPYUvQ/x8oTEuH/WFt18M5i3YK2tPg3Mypnm+uaUg1vxd+QlJRZTTTj+sSIiGflTkTMIWPpoQ9bm8/oo5MlOP630mK3/Lp117BLAqNJfUm6XW8dikhBFiWZ+IVge3bsuaIOTs+8TNmXT6F0aZZ66RQElcfdllfGSmYv39QzH95XL6d/AgSBh93he26mqlOpHf7gcsT/5UwggW8wDjyF53WYBBuYjg3A7umwHzoilLy98E3ortGF1RaMjlu0jSrlsT+pBdu1ZKwgxEfk3zqKaTYNr/0w9AudsRWMXY8F2oAOG1W5qUoabqiFiVc4AvrP2oZ6q3y1ikv1kLSjGpfmPdz524XY25xoXZOp2M5NIuVo36bZZ9tUhNtkGnE0pufzaRYDAAChWW5xvcOU1ngSDfFkogg4cUmWkVrARBhRwaUcbN2fU/lPDpdaPjg0UCLTCmO8VsGfdvX0NAGu8MUQjbOLvPqzAIR+pEDgKiRIqgW8UlByIAmjh7Z0W8NXLGcTNt2vP9qXnSJpjXP6YU7WlfFEQn9sFwIBQEiCiU2vUfV83iNoVDSaN+0zeeJCHmrY5g/FaqHvXw++vu0ZzK43ZLDy6TU58QP2qooZ6pie8/PGYWVr6U3avXtk/epC3hSHwo6tv1GA=
       on:
         tags: true
+    if: tag IS present
```
Comment (daviddavis) [inline .travis.yml]: Isn't this handled already on the previous line?

File: .travis/before_install.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+if [ -n "$PULP_PLUGIN_PR_NUMBER" ]; then
+  pushd pulpcore-plugin
+  git fetch origin +refs/pull/$PULP_PLUGIN_PR_NUMBER/merge
+  git checkout FETCH_HEAD
+  popd
+fi
+
+
+if [ -n "$PULP_SMASH_PR_NUMBER" ]; then
+  pip uninstall -y pulp-smash
```
Comment (daviddavis) [inline .travis/before_install.sh]: Why is this needed?

File: .travis.yml (hunk 1 above)
Comment (dkliban) [reply .travis.yml]: line 48 is needed. not exactly sure about line 47 though. but the docs say it's needed.

File: .travis/install.sh [hunk 3]
```diff
@@ -1,44 +1,17 @@
...
-
-git clone https://github.com/pulp/pulpcore-plugin.git
-
-if [ -n "$PULP_PLUGIN_PR_NUMBER" ]; then
-  pushd pulpcore-plugin
-  git fetch origin +refs/pull/$PULP_PLUGIN_PR_NUMBER/merge
-  git checkout FETCH_HEAD
-  popd
+if [ "$TEST" = 'docs' ]; then
+  pip3 install -r doc_requirements.txt
```
Comment (daviddavis) [inline .travis/install.sh]: Why are you using pip3 here? Asking because everywhere else uses pip.

File: .travis/before_install.sh (hunk 2 above)
Comment (dkliban) [reply .travis/before_install.sh]: copy/pasta from pulpcore. this is not needed. i'll remove.

File: .travis/install.sh (hunk 3 above)
Comment (dkliban) [reply .travis/install.sh]: copy/pasta from pulpcore. i can update.

File: .travis/script.sh [hunk 4]
```diff
@@ -1,30 +1,61 @@
...
 
 # Lint code.
-flake8 --config flake8.cfg || exit 1
-
-# Run migrations.
-export DJANGO_SETTINGS_MODULE=pulpcore.app.settings
-export PULP_CONTENT_HOST=localhost:24816
-django-admin makemigrations file
-django-admin migrate --noinput
+flake8 --config flake8.cfg
```
Comment (daviddavis) [inline .travis/script.sh]: I'm a bit hesitant about dropping the `|| exit 1` since a faster fail time would help alleviate Travis traffic. Thoughts?

Comment (dkliban) [reply .travis/script.sh]: let's fail fast

**Comment by daviddavis** [comment]
Did you intend to add testing against mariadb? Asking because I don't see it mentioned in the commit message at all.

**Comment by dkliban** [comment]
> Did you intend to add testing against mariadb? Asking because I don't see it mentioned in the commit message at all.

Yes I did mean to. I'll add it to the commit message.
